File: ViewPaper/app/src/main/java/com/example/viewpaper/MainActivity.java
package com.example.viewpaper;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.widget.EditText;

import com.google.android.material.tabs.TabLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ViewPager viewPager = findViewById(R.id.viewPaper);
        TabLayout tabLayout = findViewById(R.id.tabLayout);



        Adapter adapter= new Adapter(getSupportFragmentManager());
        adapter.add(new Log_in_Fragment(),"Log In");
        adapter.add(new Sign_Up_Fragment(),"Sign up");

        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);



    }
}

File: ContactContentProvider/app/src/main/java/com/daisymai99/contactcontentprovider/MainActivity.kt
package com.daisymai99.contactcontentprovider

import android.content.Intent
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.ContactsContract
import android.widget.Button
import android.widget.SimpleCursorAdapter
import androidx.core.app.ActivityCompat
import java.util.jar.Manifest

class MainActivity : AppCompatActivity() {




    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btn_message).setOnClickListener { v ->
            startActivity( Intent(applicationContext,Message::class.java))

        }

        findViewById<Button>(R.id.btn_Contact).setOnClickListener { v ->
            startActivity( Intent(applicationContext,Contact::class.java))

        }


    }


}